pkg/p2p/peer/peer.go

```diff
@@ -40,9 +40,16 @@ type Connection struct {
 type GossipConnector struct {
 	gossip *processing.Gossip
 	*Connection
+	protocol.ServiceFlag
 }
 
 func (g *GossipConnector) Write(b []byte) (int, error) {
+	// Preliminary check, to ensure we don't send messages to
+	// nodes who aren't interested in them.
+	if !g.canSend(b[0]) {
+		return 0, nil
+	}
```

Comment:
I would rather prefer the Gossip connector to be unaware of _whitelisting logic_ and, instead, instantiate different `GossipConnector`s depending on the peer attached. Also, checking the first byte is a hacky way to differentiate messages. If, in fact, we implement continuous streaming, the packets could be chopped off in several chunks and the first byte would not be the packet topic/category anymore

Reply:
Yep, very good point. My approach is definitely a bit hacky, but I didn't consider that option. I'll make the change

Reply:
As for reading the first byte, I'm not sure how else to approach it at the moment. One thing that came to mind is to do such a check in the `Gossip`, as I imagine it would tokenize the stream of bytes before sending it out to the peer, though I'm not sure if it's a good idea to give the `Gossip` the responsibility of deciding whether the payload is valid or not.